tests/test_examples.py

```diff
@@ -2,56 +2,56 @@
 
-from tools import test_resources_in_dataframe
-
-log_format = '[%(asctime)s] [%(levelname)s] - %(message)s'
+log_format = "[%(asctime)s] [%(levelname)s] - %(message)s"
 logging.basicConfig(level=logging.INFO, format=log_format)
 
 
-def test_1():
-    dirname = "tests/1"
-    fhir_api_url = "http://hapi.fhir.org/baseR4/"
-    path_rules = "data"
-
-    test_resources_in_dataframe(dirname, fhir_api_url, path_rules)
-
-def test_2():
-    dirname = "tests/2"
-    fhir_api_url = "http://hapi.fhir.org/baseR4/"
-    path_rules = "data"
-
-    test_resources_in_dataframe(dirname, fhir_api_url, path_rules)
-
-def test_3():
-    dirname = "tests/3"
-    fhir_api_url = "http://hapi.fhir.org/baseR4/"
-    path_rules = "data"
-
-    test_resources_in_dataframe(dirname, fhir_api_url, path_rules)
-
-def test_4():
-    dirname = "tests/4"
-    fhir_api_url = "http://hapi.fhir.org/baseR4/"
-    path_rules = "data"
-
-    test_resources_in_dataframe(dirname, fhir_api_url, path_rules)
-
-def test_5():
-    dirname = "tests/5"
-    fhir_api_url = "http://hapi.fhir.org/baseR4/"
-    path_rules = "data"
+@pytest.mark.parametrize(
+    "dirname, fhir_api_url",
+    [
+        ("tests/1", "http://hapi.fhir.org/baseR4/"),
+        ("tests/2", "http://hapi.fhir.org/baseR4/"),
+        ("tests/3", "http://hapi.fhir.org/baseR4/"),
+        ("tests/1", "http://hapi.fhir.org/baseR4/"),
+        ("tests/5", "http://hapi.fhir.org/baseR4/"),
+    ],
+)
+def test_resources_in_dataframe(dirname, fhir_api_url):
+    create_resource_test(dirname, fhir_api_url)
+    with open(os.path.join(dirname, "config.json")) as json_file:
+        config = json.load(json_file)
+    with open(os.path.join(dirname, "infos_test", "config_checks.json")) as json_file:
+        checks = json.load(json_file)
+    with open(os.path.join(dirname, "infos_test", "info_hapi.json")) as json_file:
+        info_hapi = json.load(json_file)
+
+    query = Query(fhir_api_url)
+    query.from_config(config)
+    query.execute()
+    df = query.main_dataframe
+
+    lines = checks["line"]
+    for line in lines:
+        cols = []
+        conditions = []
+        for alias, filename in line:
+            cols.append(f"{alias}:id")
+            conditions.append(info_hapi[filename])
+        condition = " & ".join(
+            [f"(df['{col}'].str.contains('{cond}'))" for col, cond in zip(cols, conditions)]
+        )
+        logging.info(condition)
+        result = df[eval(condition)]
+        logging.info(result)
+        assert len(result.index) >= 1, f"{dirname} failed"
 
-    test_resources_in_dataframe(dirname, fhir_api_url, path_rules)
 
 if __name__ == "__main__":
-    os.chdir(
-        os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
-    )
-    test_1()
-    test_2()
-    test_3()
-    test_4()
-    test_5()
+    os.chdir(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    test_resources_in_dataframe("tests/1", "http://hapi.fhir.org/baseR4/")
```

Comment:
With pytest, i don't think you need to explicitely call `test_resources_in_dataframe` it searchs automatically for all functions `def test_*`

Reply:
Indeed, this call in the __name__ == "__main__" is not necessary for `pytest` but I had put it if ever we wanted to launch the function as a classic script. Do you want me to delete it?

Reply:
I'd say the tests should from on now always be launched with pytest

Reply:
got it! i deleted it in a new commit.